### PR TITLE
[vcpkg baseline][hunspell] Fix usage when building static

### PR DESCRIPTION
**Pull request by JackBoosY**
In _include/hunspell/hunvisapi.h_:
```cpp
#ifndef HUNSPELL_VISIBILITY_H_
#define HUNSPELL_VISIBILITY_H_

#if defined(HUNSPELL_STATIC)
#  define LIBHUNSPELL_DLL_EXPORTED
#elif defined(_WIN32)
#  if defined(BUILDING_LIBHUNSPELL)
#    define LIBHUNSPELL_DLL_EXPORTED __declspec(dllexport)
#  else
#    define LIBHUNSPELL_DLL_EXPORTED __declspec(dllimport)
#  endif
#elif defined(BUILDING_LIBHUNSPELL) && 1
#  define LIBHUNSPELL_DLL_EXPORTED __attribute__((__visibility__("default")))
#else
#  define LIBHUNSPELL_DLL_EXPORTED
#endif

#endif
```
That caused the usage error in baseline:
```
hunspelldict.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl Hunspell::Hunspell(char const *,char const *,char const *)" (__imp_??0Hunspell@@QEAA@PEBD00@Z) referenced in function "public: __cdecl HunspellDict::HunspellDict(class QString const &,class QString)" (??0HunspellDict@@QEAA@AEBVQString@@V1@@Z)
hunspelldict.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl Hunspell::~Hunspell(void)" (__imp_??1Hunspell@@QEAA@XZ) referenced in function "public: void * __cdecl Hunspell::`scalar deleting destructor'(unsigned int)" (??_GHunspell@@QEAAPEAXI@Z)
hunspelldict.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: int __cdecl Hunspell::spell(char const *,int *,char * *)" (__imp_?spell@Hunspell@@QEAAHPEBDPEAHPEAPEAD@Z) referenced in function "public: virtual bool __cdecl HunspellDict::isCorrect(class QString const &)const " (?isCorrect@HunspellDict@@UEBA_NAEBVQString@@@Z)
hunspelldict.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: int __cdecl Hunspell::suggest(char * * *,char const *)" (__imp_?suggest@Hunspell@@QEAAHPEAPEAPEADPEBD@Z) referenced in function "public: virtual class QStringList __cdecl HunspellDict::suggest(class QString const &)const " (?suggest@HunspellDict@@UEBA?AVQStringList@@AEBVQString@@@Z)
hunspelldict.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: void __cdecl Hunspell::free_list(char * * *,int)" (__imp_?free_list@Hunspell@@QEAAXPEAPEAPEADH@Z) referenced in function "public: virtual class QStringList __cdecl HunspellDict::suggest(class QString const &)const " (?suggest@HunspellDict@@UEBA?AVQStringList@@AEBVQString@@@Z)
hunspelldict.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: char * __cdecl Hunspell::get_dic_encoding(void)" (__imp_?get_dic_encoding@Hunspell@@QEAAPEADXZ) referenced in function "public: __cdecl HunspellDict::HunspellDict(class QString const &,class QString)" (??0HunspellDict@@QEAA@AEBVQString@@V1@@Z)
hunspelldict.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: int __cdecl Hunspell::add(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (__imp_?add@Hunspell@@QEAAHAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) referenced in function "public: __cdecl HunspellDict::HunspellDict(class QString const &,class QString)" (??0HunspellDict@@QEAA@AEBVQString@@V1@@Z)
hunspelldict.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: int __cdecl Hunspell::add_with_affix(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (__imp_?add_with_affix@Hunspell@@QEAAHAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@0@Z) referenced in function "public: __cdecl HunspellDict::HunspellDict(class QString const &,class QString)" (??0HunspellDict@@QEAA@AEBVQString@@V1@@Z)
hunspelldict.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: int __cdecl Hunspell::remove(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (__imp_?remove@Hunspell@@QEAAHAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) referenced in function "public: __cdecl HunspellDict::HunspellDict(class QString const &,class QString)" (??0HunspellDict@@QEAA@AEBVQString@@V1@@Z)
bin\sonnet_hunspell.dll : fatal error LNK1120: 9 unresolved externals
```

### CODE REVIEW

**Review by NancyLi1013 — APPROVED** [review]

File: ports/hunspell/portfile.cmake [hunk 1]
```diff
@@ -77,6 +77,14 @@ else()
 endif()
 vcpkg_copy_pdbs()
 
+if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    vcpkg_replace_string(
+        "${CURRENT_PACKAGES_DIR}/include/hunspell/hunvisapi.h"
```
Comment (NancyLi1013) [inline ports/hunspell/portfile.cmake:82]:
```suggestion
        "${CURRENT_PACKAGES_DIR}/include/hunvisapi.h"
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: versions/h-/hunspell.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "2f2264528232cb7b66605e84634f86bbde7f2be7",
```
Comment (JackBoosY) [inline versions/h-/hunspell.json:4]:
```suggestion
      "git-tree": "2a0514dd8b3893b6d3a502fbf55156e9e971d6f7",
```

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Ping @dan-shaw, please help merge this PR, thanks.

File: ports/hunspell/portfile.cmake [hunk 3]
```diff
@@ -77,6 +77,19 @@ else()
 endif()
 vcpkg_copy_pdbs()
 
+if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    if (VCPKG_TARGET_IS_WINDOWS)
+        set(HUNSPELL_EXPORT_HDR "${CURRENT_PACKAGES_DIR}/include/hunvisapi.h")
+    else()
+        set(HUNSPELL_EXPORT_HDR "${CURRENT_PACKAGES_DIR}/include/hunspell/hunvisapi.h")
+    endif()
+    vcpkg_replace_string(
```
Comment (BillyONeal) [inline ports/hunspell/portfile.cmake:86]: It seems like it should make this be 0 if we aren't static?

Comment (JackBoosY) [reply ports/hunspell/portfile.cmake:86]: We don't need it, on dynamic it doesn't define `HUNSPELL_STATIC` and `BUILDING_LIBHUNSPELL` so we can take `__declspec(dllimport)`.

**Comment by BillyONeal** [comment]
Thanks for the fix!

File: ports/hunspell/portfile.cmake [hunk 4]
```diff
@@ -77,6 +77,19 @@ else()
 endif()
 vcpkg_copy_pdbs()
 
+if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    if (VCPKG_TARGET_IS_WINDOWS)
```
Comment (dg0yt) [inline ports/hunspell/portfile.cmake:81]: This needs a `AND NOT VCPKG_TARGET_IS_MINGW`, to match the earlier choice of build system. I just verified that mingw installs to `include/hunspell/hunvisapi.h`.

**Pull request merged**